Library/Homebrew/cmd/--env.rb

```diff
@@ -3,19 +3,37 @@
 
 require "extend/ENV"
 require "build_environment"
+require "utils/shell"
 
 module Homebrew
+  def explicit_shell
+    shell = ARGV.value("shell")
+    return :noshell if shell == "none"
```

Comment:
@UniqMartin, I like that better then what I have now, I'll make the changes. Just a few questions.

- for autodetecting the shell, do we still want to check the argv[0] of the parent process and then fall back to the value of `SHELL`? That seems to be what folks have settled on but I want to make sure.
- now that `--env` is a bona fide subcommand does it make sense to make it `brew env` (and keep the `--env` for backward compatibility).